staging-service/src/main/java/com/github/cafdataprocessing/services/staging/DiskSpaceHealthIndicatorWithTimeout.java

```diff
@@ -54,19 +64,35 @@ protected void doHealthCheck(Health.Builder builder) throws Exception
         });
         try {
             healthcheckFuture.get(healthcheckTimeoutSeconds, TimeUnit.SECONDS);
+            testWrite(path);
         } catch (final TimeoutException e) {
             healthcheckFuture.cancel(true);
             builder.down().withDetail(
                 "errorMessage",
                 String.format("Timeout after %s seconds trying to access batches directory: %s",
                               healthcheckTimeoutSeconds,
                               path.toString()));
-        } catch (final InterruptedException | ExecutionException e) {
+        } catch (final InterruptedException | ExecutionException  e) {
             healthcheckFuture.cancel(true);
             builder.down().withDetail(
                 "errorMessage",
                 String.format("Exception thrown trying to access batches directory: %s", path.toString()));
             LOGGER.warn("Exception thrown trying to access batches directory {} during healthcheck", path.toString(), e);
         }
     }
+
+    private void testWrite(final File path) throws IOException
+    {
```

Comment:
As discussed pass in Builder so you can set up or down, also build up related error message. Also do not throw exception here.